enlighten/DespikingFeature.py

```diff
@@ -0,0 +1,100 @@
+import logging
+from statistics import median
+
+import numpy as np
+
+from wasatch.ProcessedReading import ProcessedReading
+
+log = logging.getLogger(__name__)
+
+class DespikingFeature:
+
+    def __init__(self,
+                 spin_tau,
+                 spin_window):
+        self.spin_tau = spin_tau
+        self.spin_window = spin_window
+
+    def process(self, processed_reading: ProcessedReading) -> ProcessedReading:
+        # currently the only algorithm but will likely expand to accommodate more
+        return self.DarrenEtAlAlgo(processed_reading)
+
+    def DarrenEtAlAlgo(self, processed_reading: ProcessedReading) -> ProcessedReading:
+        """
+        @see Whitaker, Darren, and Kevin Hayes. 
+        "A Simple Algorithm for Despiking Raman Spectra." ChemRxiv (2018)
+        """
+        spiky_spectra = processed_reading.processed
+        log.debug(f"got spiky_spectra {spiky_spectra}")
+        tau_outlier_criteria = self.spin_tau.value()
+        window_size_m = self.spin_window.value()
+        nabla_counts = np.asarray([yt - yt_last for yt, yt_last in zip(spiky_spectra[1:],spiky_spectra[:-1])]) # 0 idx vs paper 1 idx
+        med = median(nabla_counts)
+        mad = median(np.abs(nabla_counts - med))
+        mod_z_scores = (0.6745*(nabla_counts-med))/mad
+        # nabla_counts is 1 spectra length short since its a diffing
+        # paper says ends are auto set to spiky 
+        # so just insert a 0 at the beginning that will be overwritten anyway
+        np.insert(mod_z_scores, 0, 0) 
+        mod_z_scores[0] = tau_outlier_criteria + 1
+        mod_z_scores[-1] = tau_outlier_criteria + 1
+        candidate_idxs = [idx[0] for idx, value in np.ndenumerate(mod_z_scores) if abs(value) > tau_outlier_criteria]
+        self.interpolate_zs(spiky_spectra, mod_z_scores, candidate_idxs, tau_outlier_criteria, window_size_m)
+
+        log.debug(f"despiked spectra is {spiky_spectra}")
+        processed_reading.processed = spiky_spectra
+        return processed_reading
+
+    def interpolate_zs(self, 
```

Comment:
please add a @param comment for tau summarizing typical ranges for this argument, and summarizing its relationship to the number and type of spikes squelched